README.md

```diff
@@ -1,2 +1,128 @@
+2. Argo deployment is launched with: `kubectl -n argo port-forward deployment/argo-server 2746:2746` 
+
+Artifact Repository: 
+Argo needs an artifact repository (storage for flows to run). Here we will use MinIO that we previously installed. You can use the `artifact-repositories.yaml` [in the manifests folder](manifests/artifact-repositories.yaml)
+(The secret for minio we created before comes in here). You can set it up with: `kubectl apply -f artifact-repositories.yaml`
+
+Give Argo admin roles on cluster to access MinIO: : `kubectl create rolebinding argo-default-admin --clusterrole=admin --serviceaccount=argo:default --namespace=argo` (we also had to repeat this rolebinding creating for argo and argo-server service accounts in the argo namespace. This role-binding may have to be revised in a production environment where an admin role could be problematic.
+
+### Set up an IDE (optional)
+
+If you want an IDE deployed (e.g. working on a server). Install [a vscode service](https://artifacthub.io/packages/helm/inseefrlab/vscode) via Helm, then access by port-forwarding `kubectl -n argo port-forward deployment/my-vscode 35547:8080`. You will probably have to give admin rights to this VSCode (service account: `my-vscode` over MinIO (as done for Argo)). Finally, you will also have to add Metaflow configuration variables to the ConfigMap of VSCode, for putting metaflow code packages in MinIO. 
+
+![configmap-my-vscode-configmaps3metaflow](https://github.com/SDSC-ORD/zarr-linked-data/assets/22447169/498a1306-5289-4527-9cfa-22e947d555e3)
+
+Then you just install the requirements with `pip install -r requirements.txt` and you're set to go! 
+
+# C. Example/Usage
+
+## Fake Zarr Data
+You may not have data under the zarr format yet. You can define some metadata instance objects (such as in `zarr_linked_data/data/original_data.jsonld`), then define your hierarchy levels, data level and other parameters for the creation of a random Zarr test store using `zarr_linked_data/fake_data_flow.py`.
+You can find all our example data in the `data` folder, from the original jsonld metadata instance data, to the `test_store.zarr` containing random arrays as well as the jsonld metadata.
+
+## Local scripts
+
+- Fake data creation pipeline with: `python zarr_linked_data/fake_data_flow.py`
+- Metadata consolidation Metaflow pipeline with: `python zarr_linked_data/local_dev/metadata_consolidate_metaflow.py run`
+- Retrieve URI Metaflow pipeline with: `python zarr_linked_data/local_dev/uri_matching_metaflow.py run`
+
+## Local runthrough step-by-step 
+
+Here is a detailed run through using poetry to set-up dependencies. (Please first install poetry and run `poetry install` as explained in `B. Getting Started`).
+
+1. Start your environment with `poetry shell`
+
+2. Run: `poetry run python zarr_linked_data/fake_data_flow.py`
```

Comment:
**comment:** `poetry run` executes the command inside the poetry virtualenv. With `poetry shell` before, we are already in the virtualenv and don't need it.